logic.py

```
@@ -183,475 +243,533 @@ def fmt_p_with_styling(val):
+                        interaction_cols_created.append(new_col)
+                        # Mark as linear for simplicity in display
+                        mode_map[new_col] = 'linear' 
+                    except (ValueError, TypeError) as e:
+                        logger.debug(f"Could not create interaction {new_col}: {e}")
+
+        
+        # Check for perfect separation (Logistic Only)
+        has_perfect_separation = False
+        if model_type == 'logistic' and method == 'auto':
+            for col in sorted_cols:
+                if col == outcome_name or col not in df_aligned.columns: continue
```

Ruff (0.14.10)

Multiple statements on one line (colon)

(E701)